### PR TITLE
Visualizer show classification and segmentation

### PR DESCRIPTION
**Pull request by alexriedel1**
This would be a possible fix to https://github.com/openvinotoolkit/anomalib/pull/175#issuecomment-1082988190

Alternatively, the classification image could always be added as it provides useful insights in what the model predicted.

### CODE REVIEW

File: anomalib/utils/callbacks/visualizer_callback.py [hunk 1]
```diff
@@ -111,23 +112,31 @@ def on_test_batch_end(
...
             visualizer = Visualizer(num_rows=1, num_cols=5, figure_size=(12, 3))
             visualizer.add_image(image=image, title="Image")
-            visualizer.add_image(image=true_mask, color_map="gray", title="Ground Truth")
             visualizer.add_image(image=heat_map, title="Predicted Heat Map")
             visualizer.add_image(image=pred_mask, color_map="gray", title="Predicted Mask")
             visualizer.add_image(image=vis_img, title="Segmentation Result")
+
+            if self.task == "classification":
+                image_classified = visualizer.add_text(
+                    image=image, text=f'Pred: {outputs["pred_scores"][i]:.3f} GT: {int(outputs["label"])}'
```
Comment (alexriedel1) [inline anomalib/utils/callbacks/visualizer_callback.py:132]: should be `outputs["label"][i]`

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks for the pr, would you be able to resolve the merge conflicts so we could run the CI?

**Comment by alexriedel1** [comment]
> Thanks for the pr, would you be able to resolve the merge conflicts so we could run the CI?

Ah right! Just want to note that after merging this PR https://github.com/openvinotoolkit/anomalib/pull/181 the proposed solution needs to be updated..

**Comment by samet-akcay** [comment]
@alexriedel1, the CI has been fixed now. Could you please update the PR by pulling the latest development? Thanks!

**Comment by alexriedel1** [comment]
> @alexriedel1, the CI has been fixed now. Could you please update the PR by pulling the latest development? Thanks!

done

**Comment by alexriedel1** [comment]
I merged the development branch and fixed some styling errors

**Comment by samet-akcay** [comment]
@alexriedel1, we've run the CI. There are two tests that failed unexpectedly. We'll have a look why this is the case, and let you know. I don't expect you to look the tests:)

Here is the CI log showing why it failed:
[CI-PR-178.log](https://github.com/openvinotoolkit/anomalib/files/8427540/CI-PR-178.log)

**Comment by alexriedel1** [comment]
> @alexriedel1, we've run the CI. There are two tests that failed unexpectedly. We'll have a look why this is the case, and let you know. I don't expect you to look the tests:)
> 
> Here is the CI log showing why it failed: [CI-PR-178.log](https://github.com/openvinotoolkit/anomalib/files/8427540/CI-PR-178.log)

It's because the class `VisualizerCallback` now has to be initialized with a task. I can check that and fix it if you want.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks for the efforts!

**Review by djdameln — APPROVED** [review]
Thanks for another nice contribution. I just have some minor comments

File: anomalib/utils/callbacks/visualizer_callback.py [hunk 2]
```diff
@@ -111,26 +112,33 @@ def on_test_batch_end(
...
+            visualizer.add_image(image=vis_img, title="Segmentation Result")
+
+            if self.task == "classification":
+                image_classified = visualizer.add_text(
+                    image=image, text=f'Pred: {outputs["pred_scores"][i]:.3f} GT: {int(outputs["label"][i])}'
+                )
+                visualizer.add_image(image=image_classified, title="Classified Image")
+
+            if self.task == "segmentation":
+                true_mask = outputs["mask"][i].cpu().numpy()
```
[the remark this answers is not in the document]
Comment (alexriedel1) [reply anomalib/utils/callbacks/visualizer_callback.py]: I corrected the image order but also realized that classification is always possible, so why not display the classified image in either case

File: anomalib/utils/callbacks/visualizer_callback.py [hunk 3]
```diff
@@ -111,26 +112,33 @@ def on_test_batch_end(
...
+
+            visualizer = Visualizer(num_rows=1, num_cols=5, figure_size=(12, 3))
+            visualizer.add_image(image=image, title="Image")
+            visualizer.add_image(image=heat_map, title="Predicted Heat Map")
+            visualizer.add_image(image=pred_mask, color_map="gray", title="Predicted Mask")
+            visualizer.add_image(image=vis_img, title="Segmentation Result")
+
+            if self.task == "classification":
+                image_classified = visualizer.add_text(
+                    image=image, text=f'Pred: {outputs["pred_scores"][i]:.3f} GT: {int(outputs["label"][i])}'
```
[the remark this answers is not in the document]
Comment (alexriedel1) [reply anomalib/utils/callbacks/visualizer_callback.py]: Added the proposed labels (its a pain to write on images using cv2..)

**Pull request merged**